FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +23,46 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case let .success((data, httpResponse)):
+				guard httpResponse.statusCode == Constants.successSatusCode,
+				      let images = try? JSONDecoder().decode(FeedLoaderResponse.self, from: data) else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+
+				completion(.success(images.items.feedImageItems()))
+			}
+		}
+	}
+}
+
+private struct FeedLoaderResponse: Decodable {
+	let items: [Image]
+}
+
+private struct Image: Decodable {
+	public let image_id: UUID
+	public let image_desc: String?
+	public let image_loc: String?
+	public let image_url: URL
+
+	var feedImage: FeedImage {
+		return FeedImage(
+			id: image_id,
+			description: image_desc,
+			location: image_loc,
+			url: image_url
+		)
+	}
+}
+
+extension Array where Element == Image {
```

Comment:
This `extension` is an implementation detail of `RemoteFeedLoader`, so it should be marked as `private`.
```suggestion
private extension Array where Element == Image {
```